File: Data Structure I/python/Day 7/remove_linked_list_element.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        while(head!=None and head.val==val):
            head=head.next
        temp=head
        ans=temp
        if(head!=None):
            head=head.next
        while(head!=None):
            if(head.val!=val):
                temp.next=head
                temp=temp.next
            elif(head.next==None):
                temp.next=None
            head=head.next
        return ans

File: Data Structure I/java/Day 7/RemoveLinkedListElement.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode temp;
        while(head!=null&&head.val==val){
            head=head.next;
        }
        temp=head;
        ListNode ans=temp;
        if(head!=null){
            head=head.next;
        }
        while(head!=null){
            if(head.val!=val){
                temp.next=head;
                temp=temp.next;
            }else if(head.next==null){
                temp.next=null;
            }
            head=head.next;
        }
        return ans;
    }
}